src/components/modals/SingleInputModal.js

```diff
@@ -0,0 +1,78 @@
+// @flow
```

Comment:
The general idea of using a component like this is good.
I'm just wondering, can't you get the same behavior by using the "AddressModal", maybe with some minor tweaks?

Reply:
I probably can, but

`AddressModal` is very overloaded with logic and all of it about `Address`. In my case with `Private Key` all of this code not needed, even more all of this empty space on `AddressModal` I don't need too, because this space used only for filtered `fioAddresses` which were found after the user has filled in the input.

I think we should write simple, dumb and self-documenting code which will be easy to read and maintain in future

Reply:
"I think we should write simple, dumb and self-documenting code which will be easy to read and maintain in future" 

I couldn't agree more. And you are right about not using the `AddressModal`.  I missed a lot of things regarding that component.
But, part of having "easy to maintain" code, is also not having duplicated, almost identical components.
And if you kept looking in the `modals` folder, you would have found the `SendLogsModal`.

That one is almost identical to the one you are using here.
But at this point, since you already wrote the component, you might as well replace the `SendLogsModal` with the modal you wrote. Or the other way around, but do try and merge them.